### PR TITLE
fix: Unreadable Asset options

### LINKED ISSUE #11376: In dark mode, the token details action menu has dark text as well, making the text not readable.
### What is this about?

In dark mode, the token details action menu has dark text as well, making the text not readable. Ideally, in dark mode, the text should be white, just like the open icon to the left.

image

### Scenario

WHEN a user is in the token details view
AND the user clicks on the three dots
AND the user is in dark mode
The text is black instead of white.

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [X] Design
- [X] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

On darkmode, asset options in bottom drawer were unreabable.

Additionally, when pressing on new "balance" list item, the button didn't do anything. This PR also adds the navigation to TokenDetails.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/11376

## **Manual testing steps**

1. Toggle dark mode in system settings
2. Navigate to token detail page
3. Press ellipsis in top right to open drawer. Text should be white on darkmode, black on light mode
4. Additionally, when pressing balance from token detail screen, it should be pressable and navigate to the token detail view. This should not navigate anywhere for native gas token, and only for erc20 tokens on each network (DAI, USDC, SushiSwap, etc)

### **Before**

IMG_26837C390124-1

### **After**

https://github.com/user-attachments/assets/52d73fea-81ef-4f26-bdfa-cfc8b198b44a

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Cal-L — DISMISSED** [review]
LGTM

File: app/components/UI/AssetOverview/Balance/index.test.tsx
Comment (NicolasMassart) [inline app/components/UI/AssetOverview/Balance/index.test.tsx:1]: Q: could you add a unit test for non regression?
(on press + check mock nav called)

Comment (gambinish) [reply app/components/UI/AssetOverview/Balance/index.test.tsx:1]: Sure thing, added here: https://github.com/MetaMask/metamask-mobile/pull/11380/commits/a4861aec9b18bdb2eb267b218328cf5bae7c2f2b

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**